### PR TITLE
[nginx-ingress-controller] cleanup before update updating nginx controller to 0.7

### LINKED ISSUE #884: SSL as a Reverse Proxy on Nginx Controller ?
Hi, i'm using the nginx and GCE official ingress controllers, but the reverse fot the backend service  is http only, it is posible a proxy pass with https support or maybe using the redirect.

` proxy_redirect      http://localhost:8080 https://secure.domain.com;`

### LINKED ISSUE #930: Nginx Ingress Controller - Updating Nginx ConfigMap doesn't trigger nginx.conf update
If I push an update to nginx's ConfigMap resource, it doesn't seem the nginx.conf is updated anytime "soon." I have to push a bogus ingress change to force it to happen. Is there a better way to get it to recognize ConfigMap changes?

### PR DESCRIPTION
**Pull request by aledbf**
fixes #930
fixes #884

### CODE REVIEW

File: ingress/controllers/nginx/nginx/secureupstream/main.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import (
+	"errors"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+const (
+	secureUpstream = "ingress.kubernetes.io/secure-upstream"
```
Comment (bprashanth) [inline ingress/controllers/nginx/nginx/secureupstream/main.go]: please add a comment with explanantion, or link to docs about how the encryption is performed to the upstream (which certs are used etc).

File: ingress/controllers/nginx/controller.go [hunk 2]
```diff
@@ -89,9 +90,11 @@ type loadBalancerController struct {
 	ingController  *framework.Controller
 	endpController *framework.Controller
 	svcController  *framework.Controller
+	mapController  *framework.Controller
 	ingLister      StoreToIngressLister
 	svcLister      cache.StoreToServiceLister
 	endpLister     cache.StoreToEndpointsLister
+	mapLister      StoreToMapLister
```
Comment (bprashanth) [inline ingress/controllers/nginx/controller.go]: can you rename to storetoconfigmaplister? since this makes it sounds like you're listing to a go map

Comment (aledbf) [inline ingress/controllers/nginx/controller.go]: done

**Comment by bprashanth** [comment]
LGTM, guess we'll need followups for examples and readme anyway so you can fix the nits in that

File: ingress/controllers/nginx/README.md [hunk 3]
```diff
@@ -260,239 +258,6 @@ In case of an error in a request the body of the response is obtained from the `
 Using this two headers is possible to use a custom backend service like [this one](https://github.com/aledbf/contrib/tree/nginx-debug-server/Ingress/images/nginx-error-server) that inspect each request and returns a custom error page with the format expected by the client. Please check the example [custom-errors](examples/custom-errors/README.md)
 
 
-### Annotations
```
Comment (databus23) [inline ingress/controllers/nginx/README.md]: Was this deletion of documentation by accident? The README is no missing vital information about the available configuration options.

Comment (aledbf) [inline ingress/controllers/nginx/README.md]: @databus23 this is being added here https://github.com/kubernetes/contrib/pull/1130

**Pull request merged**